src/lib/tradingView/dydxfeed/utils.ts

```diff
@@ -0,0 +1,106 @@
+import { TOKEN_DECIMALS } from '@/constants/numbers';
+import { ThemeColorBase } from '@/constants/styles/colors';
+
+import { formatNumberOutput, OutputType } from '@/components/Output';
+
+import { RootStore } from '@/state/_store';
+import { getFillDetails } from '@/state/accountSelectors';
+
+import { getAverageFillPrice } from '@/lib/orders';
+
+/**
+ * @description Converts times in ms to the appropriate bar time (in seconds)
+ * For example, if the starting time = 5000ms, interval = 10,000ms, a value of 26,000ms would
+ * be grouped into the bar at 25,000ms = 25s
+ */
+export function getBarTime(
+  chartStartTimeMs: number,
+  fillTimeMs: number,
+  resolution: ResolutionString
+): number | undefined {
+  const intervalMs = RESOLUTION_TO_INTERVAL_MS[resolution];
+
+  const [startBn, intervalSizeBn, fillTimeBn] = [
+    BigNumber(chartStartTimeMs),
+    BigNumber(intervalMs),
+    BigNumber(fillTimeMs),
+  ];
+  const numIntervalsBetween = fillTimeBn.minus(startBn).dividedToIntegerBy(intervalSizeBn);
+  return startBn
+    .plus(numIntervalsBetween.multipliedBy(intervalSizeBn))
+    .dividedToIntegerBy(1000)
+    .toNumber();
+}
+
+export const getMarkForOrderFills = (
+  store: RootStore,
+  orderFills: SubaccountFills,
+  orderId: string,
+  barStartMs: number,
+  resolution: ResolutionString,
+  stringGetter: StringGetterFunction,
+  localeSeparators: { group?: string; decimal?: string },
+  selectedLocale: SupportedLocales,
+  theme: ThemeColorBase
+): Mark => {
+  const formattedAveragePrice = formatNumberOutput(
+    getAverageFillPrice(orderFills),
+    OutputType.Fiat,
+    {
+      decimalSeparator: localeSeparators.decimal,
+      groupSeparator: localeSeparators.group,
+      selectedLocale,
+    }
+  );
+  const formattedSize = formatNumberOutput(
+    sum(orderFills.map((fill) => fill.size)),
+    OutputType.Asset,
+    {
+      fractionDigits: TOKEN_DECIMALS,
+      decimalSeparator: localeSeparators.decimal,
+      groupSeparator: localeSeparators.group,
+      selectedLocale,
+    }
+  );
+  const fill = orderFills[0];
+  const fillDetails = getFillDetails()(store.getState(), fill.id);
+
+  const textParams = {
+    ASSET_SIZE: formattedSize,
+    ASSET: fillDetails?.asset?.id,
+    PRICE: formattedAveragePrice,
+  };
+
+  const text = stringGetter({
+    key:
+      fill.side.name === AbacusOrderSide.Buy.name
+        ? STRING_KEYS.BUY_MARK_TOOLTIP
+        : STRING_KEYS.SELL_MARK_TOOLTIP,
+    params: textParams,
+  }) as string;
+
+  const color: MarkCustomColor =
+    fill.side.name === AbacusOrderSide.Buy.name
+      ? {
+          color: theme.positive,
+          background: theme.positive,
+        }
+      : { color: theme.negative, background: theme.negative };
+
+  return {
+    id: orderId,
+    time: getBarTime(barStartMs, fill.createdAtMilliseconds, resolution) ?? 0,
+    minSize: 16,
+    text,
+    labelFontColor: 'white',
```

Comment:
could be follow up, but instead of white, can we use one of the `text-color-x` variables? (to have it better align with themeing)! also nit, but any chance we can also update the border color to better align?

<img width="788" alt="Screenshot 2024-08-16 at 9 36 48 AM" src="https://github.com/user-attachments/assets/5ef206bf-f865-49fb-a9b9-1641b93cd32d">

Reply:
ok unfort theres no exposed way for us to change the border color until we upgrade the library but i can add that to the TODO!

will update the text color based on theme tho :D